### PR TITLE
[SPIRV] Fix SPV_KHR_expect_assume support

### PR DESCRIPTION
**Pull request by pmatos**
Since efe0e10718 changes in tests are required. Need to add extension to Extensions list
and command line to enable use of the extension for test runs.

### CODE REVIEW

**Comment by pmatos** [comment]
This is a draft because I noticed that if we do not enable the use of the extension, extension and capability are not declared but instructions are still generated. What's the correct behaviour for this? I would think that since assume and expect do not alter the semantics we can just not issue them. The alternative is to generate an error.

What do you think? Shall we just ignore the instruction if we cannot use the extension?

**Comment by iliya-diyachkov** [comment]
SPIRV translator do not generate OpAssumeTrueKHR and OpExpectKHR instructions without an error message. I think we should do the same.

File: llvm/lib/Target/SPIRV/SPIRVPrepareFunctions.cpp [hunk 1]
```diff
@@ -376,6 +380,7 @@ SPIRVPrepareFunctions::removeAggregateTypesFromSignature(Function *F) {
 
 bool SPIRVPrepareFunctions::runOnModule(Module &M) {
   bool Changed = false;
+
```
Comment (iliya-diyachkov) [inline llvm/lib/Target/SPIRV/SPIRVPrepareFunctions.cpp:383]: Do we really need this line?

**Review by iliya-diyachkov — APPROVED** [review]
Otherwise, it looks good to me.

File: llvm/lib/Target/SPIRV/SPIRVPrepareFunctions.cpp (hunk 1 above)
Comment (pmatos) [reply llvm/lib/Target/SPIRV/SPIRVPrepareFunctions.cpp:383]: The newline?
No, not really. Removing and landing. Thanks.

**Pull request merged**